File: VS2005/TimerComputerShutdown/WindowsController/VB.NET/TestApp.vb
Option Explicit On 
Option Strict On

Imports Org.Mentalis.Utilities
Imports Microsoft.VisualBasic

Public Class TestApp
	Public Shared Sub Main()
		Console.WriteLine("Restart Program\r\n--------------\r\n\r\n Select the restart mode:\r\n   l - LogOff\r\n   p - PowerOff\r\n   r - Reboot\r\n   s - Shutdown\r\n   u - Suspend\r\n   h - Hibernate\r\n   c - Exit the program\r\n\r\n Make your choice: ".Replace("\r\n", ControlChars.CrLf))
		Dim input As String = Console.ReadLine()
		Select Case input.ToLower()
			Case "l"
				WindowsController.ExitWindows(RestartOptions.LogOff, False)
			Case "p"
				WindowsController.ExitWindows(RestartOptions.PowerOff, False)
			Case "r"
				WindowsController.ExitWindows(RestartOptions.Reboot, False)
			Case "s"
				WindowsController.ExitWindows(RestartOptions.ShutDown, False)
			Case "u"
				WindowsController.ExitWindows(RestartOptions.Suspend, False)
			Case "h"
				WindowsController.ExitWindows(RestartOptions.Hibernate, False)
			Case "c"
				Return
			Case Else
				Console.WriteLine("Invalid input...")
		End Select
	End Sub
End Class

File: trunk/VS2005/TimerComputerShutdown/WindowsController/C#/TestApp.cs
using System;
using Org.Mentalis.Utilities;

public class TestApp {
	public static void Main() {
		Console.WriteLine("Restart Program\r\n--------------\r\n\r\n Select the restart mode:\r\n   l - LogOff\r\n   p - PowerOff\r\n   r - Reboot\r\n   s - Shutdown\r\n   u - Suspend\r\n   h - Hibernate\r\n   c - Exit the program\r\n\r\n Make your choice: ");
		string input = Console.ReadLine();
		switch(input.ToLower()) {
			case "l":
				WindowsController.ExitWindows(RestartOptions.LogOff, false);
				break;
			case "p":
				WindowsController.ExitWindows(RestartOptions.PowerOff, false);
				break;
			case "r":
				WindowsController.ExitWindows(RestartOptions.Reboot, false);
				break;
			case "s":
				WindowsController.ExitWindows(RestartOptions.ShutDown, false);
				break;
			case "u":
				WindowsController.ExitWindows(RestartOptions.Suspend, false);
				break;
			case "h":
				WindowsController.ExitWindows(RestartOptions.Hibernate, false);
				break;
			case "c":
				return;
				break;
			default:
				Console.WriteLine("Invalid input...");
				break;
		}
	}
}